Move duplicated methods from MediaSourcePrivate/SourceBufferPrivate to base class

jyavenard opened the pull request:
#### e25a76a6050724c85a623b23746c2a143f844be3
<pre>
Move duplicated methods from MediaSourcePrivate/SourceBufferPrivate to base class
<a href="https://bugs.webkit.org/show_bug.cgi?id=264142">https://bugs.webkit.org/show_bug.cgi?id=264142</a>
<a href="https://rdar.apple.com/117892834">rdar://117892834</a>

Reviewed by Jer Noble.

We have three MSE implementations in our tree:
- Cocoa/AVFObjC
- Mock
- GStreamer

In a lot of instances, all three duplicate the same code. It makes it harder to maintain and too easy to introduce discrepancy between the three implementations.

So we move to their respective base classes what can be moved.

Covered by existing tests.

* Source/WebCore/platform/graphics/MediaSourcePrivate.cpp:
(WebCore::MediaSourcePrivate::~MediaSourcePrivate):
(WebCore::MediaSourcePrivate::removeSourceBuffer):
(WebCore::MediaSourcePrivate::sourceBufferPrivateDidChangeActiveState):
(WebCore::MediaSourcePrivate::hasAudio const):
(WebCore::MediaSourcePrivate::hasVideo const):
(WebCore::MediaSourcePrivate::setCDMSession):
* Source/WebCore/platform/graphics/MediaSourcePrivate.h:
(WebCore::MediaSourcePrivate::bufferedChanged): Deleted.
(WebCore::MediaSourcePrivate::setTimeFudgeFactor): Deleted.
(WebCore::MediaSourcePrivate::timeFudgeFactor const): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::SourceBufferPrivate):
(WebCore::SourceBufferPrivate::removedFromMediaSource):
(WebCore::SourceBufferPrivate::currentMediaTime const):
(WebCore::SourceBufferPrivate::duration const):
(WebCore::SourceBufferPrivate::setActive):
* Source/WebCore/platform/graphics/SourceBufferPrivate.h:
(WebCore::SourceBufferPrivate::clearMediaSource):
(WebCore::SourceBufferPrivate::didReceiveInitializationSegment const):
(WebCore::SourceBufferPrivate::setCDMSession):
(WebCore::SourceBufferPrivate::setCDMInstance):
(WebCore::SourceBufferPrivate::waitingForKey const):
(WebCore::SourceBufferPrivate::attemptToDecrypt):
(WebCore::SourceBufferPrivate::isActive const):
(WebCore::SourceBufferPrivate::isSeeking const):
(WebCore::SourceBufferPrivate::currentMediaTime const): Deleted.
(WebCore::SourceBufferPrivate::duration const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::shouldEnsureLayer const):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::setCDMSession):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaSourcePrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaSourcePrivateAVFObjC.mm:
(WebCore::MediaSourcePrivateAVFObjC::MediaSourcePrivateAVFObjC):
(WebCore::MediaSourcePrivateAVFObjC::~MediaSourcePrivateAVFObjC):
(WebCore::MediaSourcePrivateAVFObjC::addSourceBuffer):
(WebCore::MediaSourcePrivateAVFObjC::removeSourceBuffer):
(WebCore::MediaSourcePrivateAVFObjC::notifyActiveSourceBuffersChanged):
(WebCore::MediaSourcePrivateAVFObjC::markEndOfStream):
(WebCore::MediaSourcePrivateAVFObjC::hasSelectedVideo const):
(WebCore::MediaSourcePrivateAVFObjC::willSeek):
(WebCore::MediaSourcePrivateAVFObjC::naturalSize const):
(WebCore::MediaSourcePrivateAVFObjC::flushActiveSourceBuffersIfNeeded):
(WebCore::MediaSourcePrivateAVFObjC::needsVideoLayer const):
(WebCore::MediaSourcePrivateAVFObjC::unmarkEndOfStream): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::sourceBufferPrivateDidChangeActiveState): Deleted.
(WebCore::MediaSourcePrivateAVFObjCHasAudio): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::hasAudio const): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::hasVideo const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.h:
(isType):
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::SourceBufferPrivateAVFObjC::create):
(WebCore::SourceBufferPrivateAVFObjC::SourceBufferPrivateAVFObjC):
(WebCore::m_logIdentifier):
(WebCore::SourceBufferPrivateAVFObjC::didProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferPrivateAVFObjC::trackDidChangeSelected):
(WebCore::SourceBufferPrivateAVFObjC::trackDidChangeEnabled):
(WebCore::SourceBufferPrivateAVFObjC::setCDMSession):
(WebCore::SourceBufferPrivateAVFObjC::outputObscuredDueToInsufficientExternalProtectionChanged):
(WebCore::SourceBufferPrivateAVFObjC::player const):
(WebCore::SourceBufferPrivateAVFObjC::removedFromMediaSource): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::setActive): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::isActive const): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::currentMediaTime const): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::duration const): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.cpp:
(WebCore::MediaSourcePrivateGStreamer::~MediaSourcePrivateGStreamer):
(WebCore::MediaSourcePrivateGStreamer::addSourceBuffer):
(WebCore::MediaSourcePrivateGStreamer::markEndOfStream):
(WebCore::MediaSourcePrivateGStreamer::startPlaybackIfHasAllTracks):
(WebCore::MediaSourcePrivateGStreamer::removeSourceBuffer): Deleted.
(WebCore::MediaSourcePrivateGStreamer::unmarkEndOfStream): Deleted.
(WebCore::MediaSourcePrivateGStreamer::sourceBufferPrivateDidChangeActiveState): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::create):
(WebCore::SourceBufferPrivateGStreamer::SourceBufferPrivateGStreamer):
(WebCore::SourceBufferPrivateGStreamer::removedFromMediaSource):
(WebCore::SourceBufferPrivateGStreamer::flush):
(WebCore::SourceBufferPrivateGStreamer::didReceiveInitializationSegment):
(WebCore::SourceBufferPrivateGStreamer::setActive): Deleted.
(WebCore::SourceBufferPrivateGStreamer::isActive const): Deleted.
(WebCore::SourceBufferPrivateGStreamer::currentMediaTime const): Deleted.
(WebCore::SourceBufferPrivateGStreamer::duration const): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h:
(isType):
* Source/WebCore/platform/mock/mediasource/MockMediaSourcePrivate.cpp:
(WebCore::MockMediaSourcePrivate::addSourceBuffer):
(WebCore::MockMediaSourcePrivate::duration const):
(WebCore::MockMediaSourcePrivate::markEndOfStream):
(WebCore::MockMediaSourcePrivate::notifyActiveSourceBuffersChanged):
(WebCore::MockMediaSourcePrivate::~MockMediaSourcePrivate): Deleted.
(WebCore::MockMediaSourcePrivate::removeSourceBuffer): Deleted.
(WebCore::MockMediaSourcePrivate::duration): Deleted.
(WebCore::MockMediaSourcePrivate::unmarkEndOfStream): Deleted.
(WebCore::MockMediaSourcePrivate::sourceBufferPrivateDidChangeActiveState): Deleted.
(WebCore::MockSourceBufferPrivateHasAudio): Deleted.
(WebCore::MockMediaSourcePrivate::hasAudio const): Deleted.
(WebCore::MockSourceBufferPrivateHasVideo): Deleted.
(WebCore::MockMediaSourcePrivate::hasVideo const): Deleted.
* Source/WebCore/platform/mock/mediasource/MockMediaSourcePrivate.h:
* Source/WebCore/platform/mock/mediasource/MockSourceBufferPrivate.cpp:
(WebCore::MockSourceBufferPrivate::create):
(WebCore::MockSourceBufferPrivate::MockSourceBufferPrivate):
(WebCore::MockSourceBufferPrivate::mediaSourcePrivate const):
(WebCore::MockSourceBufferPrivate::readyState const):
(WebCore::MockSourceBufferPrivate::setReadyState):
(WebCore::MockSourceBufferPrivate::enqueueSample):
(WebCore::MockSourceBufferPrivate::removedFromMediaSource): Deleted.
(WebCore::MockSourceBufferPrivate::setActive): Deleted.
(WebCore::MockSourceBufferPrivate::isActive const): Deleted.
(WebCore::MockSourceBufferPrivate::currentMediaTime const): Deleted.
(WebCore::MockSourceBufferPrivate::duration const): Deleted.
* Source/WebCore/platform/mock/mediasource/MockSourceBufferPrivate.h:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::markEndOfStream):
(WebKit::MediaSourcePrivateRemote::unmarkEndOfStream):
(WebKit::MediaSourcePrivateRemote::isEnded const): Deleted.
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
(WebKit::SourceBufferPrivateRemote::create):
(WebKit::SourceBufferPrivateRemote::SourceBufferPrivateRemote):
(WebKit::SourceBufferPrivateRemote::setReadyState):
(WebKit::SourceBufferPrivateRemote::setActive):
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.h:

Canonical link: <a href="https://commits.webkit.org/270354@main">https://commits.webkit.org/270354@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/25c7c4feaf01fe06dcf2b648c8ab80506cd88274

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25196 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3737 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26452 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27311 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23120 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/25465 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5450 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1173 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23356 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25439 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2761 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21742 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27890 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2446 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22682 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28805 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22993 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23030 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26631 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2399 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/687 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3747 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6051 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2836 "Built successfully") | | [⏳ 🛠 jsc-mips ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2731 "Built successfully") | | [⏳ 🧪 jsc-mips-tests ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
<!--EWS-Status-Bubble-End-->

jernoble reviewed (approved):
r=me once the bots are happy.

jyavenard committed:
Move duplicated methods from MediaSourcePrivate/SourceBufferPrivate to base class
https://bugs.webkit.org/show_bug.cgi?id=264142
rdar://117892834

Reviewed by Jer Noble.

We have three MSE implementations in our tree:
- Cocoa/AVFObjC
- Mock
- GStreamer

In a lot of instances, all three duplicate the same code. It makes it harder to maintain and too easy to introduce discrepancy between the three implementations.

So we move to their respective base classes what can be moved.

Covered by existing tests.

* Source/WebCore/platform/graphics/MediaSourcePrivate.cpp:
(WebCore::MediaSourcePrivate::~MediaSourcePrivate):
(WebCore::MediaSourcePrivate::removeSourceBuffer):
(WebCore::MediaSourcePrivate::sourceBufferPrivateDidChangeActiveState):
(WebCore::MediaSourcePrivate::hasAudio const):
(WebCore::MediaSourcePrivate::hasVideo const):
(WebCore::MediaSourcePrivate::setCDMSession):
* Source/WebCore/platform/graphics/MediaSourcePrivate.h:
(WebCore::MediaSourcePrivate::bufferedChanged): Deleted.
(WebCore::MediaSourcePrivate::setTimeFudgeFactor): Deleted.
(WebCore::MediaSourcePrivate::timeFudgeFactor const): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::SourceBufferPrivate):
(WebCore::SourceBufferPrivate::removedFromMediaSource):
(WebCore::SourceBufferPrivate::currentMediaTime const):
(WebCore::SourceBufferPrivate::duration const):
(WebCore::SourceBufferPrivate::setActive):
* Source/WebCore/platform/graphics/SourceBufferPrivate.h:
(WebCore::SourceBufferPrivate::clearMediaSource):
(WebCore::SourceBufferPrivate::didReceiveInitializationSegment const):
(WebCore::SourceBufferPrivate::setCDMSession):
(WebCore::SourceBufferPrivate::setCDMInstance):
(WebCore::SourceBufferPrivate::waitingForKey const):
(WebCore::SourceBufferPrivate::attemptToDecrypt):
(WebCore::SourceBufferPrivate::isActive const):
(WebCore::SourceBufferPrivate::isSeeking const):
(WebCore::SourceBufferPrivate::currentMediaTime const): Deleted.
(WebCore::SourceBufferPrivate::duration const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::shouldEnsureLayer const):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::setCDMSession):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaSourcePrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaSourcePrivateAVFObjC.mm:
(WebCore::MediaSourcePrivateAVFObjC::MediaSourcePrivateAVFObjC):
(WebCore::MediaSourcePrivateAVFObjC::~MediaSourcePrivateAVFObjC):
(WebCore::MediaSourcePrivateAVFObjC::addSourceBuffer):
(WebCore::MediaSourcePrivateAVFObjC::removeSourceBuffer):
(WebCore::MediaSourcePrivateAVFObjC::notifyActiveSourceBuffersChanged):
(WebCore::MediaSourcePrivateAVFObjC::markEndOfStream):
(WebCore::MediaSourcePrivateAVFObjC::hasSelectedVideo const):
(WebCore::MediaSourcePrivateAVFObjC::willSeek):
(WebCore::MediaSourcePrivateAVFObjC::naturalSize const):
(WebCore::MediaSourcePrivateAVFObjC::flushActiveSourceBuffersIfNeeded):
(WebCore::MediaSourcePrivateAVFObjC::needsVideoLayer const):
(WebCore::MediaSourcePrivateAVFObjC::unmarkEndOfStream): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::sourceBufferPrivateDidChangeActiveState): Deleted.
(WebCore::MediaSourcePrivateAVFObjCHasAudio): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::hasAudio const): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::hasVideo const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.h:
(isType):
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::SourceBufferPrivateAVFObjC::create):
(WebCore::SourceBufferPrivateAVFObjC::SourceBufferPrivateAVFObjC):
(WebCore::m_logIdentifier):
(WebCore::SourceBufferPrivateAVFObjC::didProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferPrivateAVFObjC::trackDidChangeSelected):
(WebCore::SourceBufferPrivateAVFObjC::trackDidChangeEnabled):
(WebCore::SourceBufferPrivateAVFObjC::setCDMSession):
(WebCore::SourceBufferPrivateAVFObjC::outputObscuredDueToInsufficientExternalProtectionChanged):
(WebCore::SourceBufferPrivateAVFObjC::player const):
(WebCore::SourceBufferPrivateAVFObjC::removedFromMediaSource): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::setActive): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::isActive const): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::currentMediaTime const): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::duration const): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.cpp:
(WebCore::MediaSourcePrivateGStreamer::~MediaSourcePrivateGStreamer):
(WebCore::MediaSourcePrivateGStreamer::addSourceBuffer):
(WebCore::MediaSourcePrivateGStreamer::markEndOfStream):
(WebCore::MediaSourcePrivateGStreamer::startPlaybackIfHasAllTracks):
(WebCore::MediaSourcePrivateGStreamer::removeSourceBuffer): Deleted.
(WebCore::MediaSourcePrivateGStreamer::unmarkEndOfStream): Deleted.
(WebCore::MediaSourcePrivateGStreamer::sourceBufferPrivateDidChangeActiveState): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::create):
(WebCore::SourceBufferPrivateGStreamer::SourceBufferPrivateGStreamer):
(WebCore::SourceBufferPrivateGStreamer::removedFromMediaSource):
(WebCore::SourceBufferPrivateGStreamer::flush):
(WebCore::SourceBufferPrivateGStreamer::didReceiveInitializationSegment):
(WebCore::SourceBufferPrivateGStreamer::setActive): Deleted.
(WebCore::SourceBufferPrivateGStreamer::isActive const): Deleted.
(WebCore::SourceBufferPrivateGStreamer::currentMediaTime const): Deleted.
(WebCore::SourceBufferPrivateGStreamer::duration const): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h:
(isType):
* Source/WebCore/platform/mock/mediasource/MockMediaSourcePrivate.cpp:
(WebCore::MockMediaSourcePrivate::addSourceBuffer):
(WebCore::MockMediaSourcePrivate::duration const):
(WebCore::MockMediaSourcePrivate::markEndOfStream):
(WebCore::MockMediaSourcePrivate::notifyActiveSourceBuffersChanged):
(WebCore::MockMediaSourcePrivate::~MockMediaSourcePrivate): Deleted.
(WebCore::MockMediaSourcePrivate::removeSourceBuffer): Deleted.
(WebCore::MockMediaSourcePrivate::duration): Deleted.
(WebCore::MockMediaSourcePrivate::unmarkEndOfStream): Deleted.
(WebCore::MockMediaSourcePrivate::sourceBufferPrivateDidChangeActiveState): Deleted.
(WebCore::MockSourceBufferPrivateHasAudio): Deleted.
(WebCore::MockMediaSourcePrivate::hasAudio const): Deleted.
(WebCore::MockSourceBufferPrivateHasVideo): Deleted.
(WebCore::MockMediaSourcePrivate::hasVideo const): Deleted.
* Source/WebCore/platform/mock/mediasource/MockMediaSourcePrivate.h:
* Source/WebCore/platform/mock/mediasource/MockSourceBufferPrivate.cpp:
(WebCore::MockSourceBufferPrivate::create):
(WebCore::MockSourceBufferPrivate::MockSourceBufferPrivate):
(WebCore::MockSourceBufferPrivate::mediaSourcePrivate const):
(WebCore::MockSourceBufferPrivate::readyState const):
(WebCore::MockSourceBufferPrivate::setReadyState):
(WebCore::MockSourceBufferPrivate::enqueueSample):
(WebCore::MockSourceBufferPrivate::removedFromMediaSource): Deleted.
(WebCore::MockSourceBufferPrivate::setActive): Deleted.
(WebCore::MockSourceBufferPrivate::isActive const): Deleted.
(WebCore::MockSourceBufferPrivate::currentMediaTime const): Deleted.
(WebCore::MockSourceBufferPrivate::duration const): Deleted.
* Source/WebCore/platform/mock/mediasource/MockSourceBufferPrivate.h:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::markEndOfStream):
(WebKit::MediaSourcePrivateRemote::unmarkEndOfStream):
(WebKit::MediaSourcePrivateRemote::isEnded const): Deleted.
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
(WebKit::SourceBufferPrivateRemote::create):
(WebKit::SourceBufferPrivateRemote::SourceBufferPrivateRemote):
(WebKit::SourceBufferPrivateRemote::setReadyState):
(WebKit::SourceBufferPrivateRemote::setActive):
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.h:

Canonical link: https://commits.webkit.org/270354@main
